bin/live/pycbc_live_combine_dq_trigger_rates

```diff
@@ -0,0 +1,103 @@
+#!/usr/bin/python
+
+# Copyright 2024 Max Trevor
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License as published by the
+# Free Software Foundation; either version 3 of the License, or (at your
+# option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+# Public License for more details.
+
+"""Combine the data-quality adjusted trigger rates from multiple days."""
+
+import logging
+import argparse
+
+import numpy as np
+import h5py as h5
+
+import pycbc
+
+parser = argparse.ArgumentParser(description=__doc__)
+pycbc.add_common_pycbc_options(parser)
+parser.add_argument("--daily-dq-files", nargs="+", required=True,
+                    help="Files containing daily dq trigger rates")
+parser.add_argument("--ifo", required=True)
+parser.add_argument("--output", required=True)
+args = parser.parse_args()
+
+pycbc.init_logging(args.verbose)
+
+daily_files = args.daily_dq_files
+daily_files.sort()
+
+# need all files to use compatible settings
+# get settings hash from the last file
+# we will only use files that have the same hash
+with h5.File(daily_files[-1], 'r') as last_file:
+    settings_hash = last_file.attrs['settings_hash']
+    bin_str = last_file.attrs['background_bins']
+    bank_file = last_file.attrs['bank_file']
+    f_lower = last_file.attrs['f_lower']
+    dq_thresh = last_file.attrs['dq_thresh']
+    dq_channel = last_file.attrs['dq_channel']
+    dq_ok_channel = last_file.attrs['dq_ok_channel']
+    bin_group = last_file[f'{args.ifo}/bins']
+    template_bins = {bin_name: bin_group[bin_name]['tids'][:]
+                     for bin_name in bin_group.keys()}
+num_bins = len(bin_str.split())
+
+total_livetime = 0
+flagged_livetime = 0
+total_triggers = np.zeros(num_bins)
+flagged_triggers = np.zeros(num_bins)
+for fpath in daily_files:
+    with h5.File(fpath, 'r') as f:
+        if f.attrs['settings_hash'] != settings_hash:
+            warning_str = f'File {fpath} has incompatible settings, skipping'
+            logging.warning(warning_str)
+            continue
```

Comment:
I'm tempted to suggest to raise an error here, but I am sure there is a reason not to

Reply:
Yes, we want to have the flexibility to change the settings mid-run.
If we adjust the settings mid-run, this way will just ignore the files with old settings. Having an error would prevent this job from being able to finish after a settings change.